ocaml/lambda/translcore.ml

```diff
@@ -1242,6 +1269,8 @@ and transl_curried_function
         let arg_mode = transl_alloc_mode arg_mode in
         let params = {
           name = param ;
+          var_uid = Ident.Tbl.find_opt (uids_of_vars cases) param
+                    |> Option.value ~default:Uid.internal_not_actually_unique ;
```

Comment:
This seems expensive and unnecessary.  I think where `param` is calculated in `typecore`, the uid is available.  That is, `Typecore.name_pattern` has access to the relevant uid in the case it doesn't use the default, and we should just rememeber that uid as part of the `Texp_function` rather than doing all this work here.